### PR TITLE
add typescript .d.ts

### PR DESCRIPTION
**Pull request by vincentneo**
This .d.ts file should help improve compatibility for TypeScript users.

Note: I have never dealt with the making-of or contributing for any npm packages so I don't know if what I'm doing here is correct, but the .d.ts file should be good enough for use, as I'm already using it in my own project right now.

### CODE REVIEW

File: src/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+
+interface Props<T> extends DOMAttributes<T> {
+    children: JSX.Element;
+    component?: string;
+    duration?: number;
+    disableInitialAnimation?: boolean;
+    className: string;
```
Comment (asbjornh) [inline src/index.d.ts]: Thanks for the PR! `className` should be optional I think

Comment (vincentneo) [reply src/index.d.ts]: I think so too. Will update that.

**Review by asbjornh — APPROVED** [review]

**Comment by asbjornh** [comment]
I'll try to remember to release this soon. You're welcome to remind me if I forget it :D

**Comment by asbjornh** [comment]
Published! Thanks for your contribution :)

**Pull request merged**